source/user/sgx.rst

```diff
@@ -0,0 +1,52 @@
+.. _user_sgx:
+
+Seminumerical Exchange (SGX)
+****************************
+
+*Modules*: :mod:`sgx`
+
+The SGX module implements seminumerical computation of the exchange matrix.
+
+Introduction
+============
+
+Direct computation of the Hartree-Fock Exchange matrix in the atomic orbital basis scales poorly with system size. To achieve better scaling, one coordinate of the two-electron integrals is computed analytically, while the other is evaluated on a real-space grid, as proposed by Friesner :cite:`Friesner1985`. The PySCF implementation most closely resembles the chain-of-spheres (COSX) algorithm of Neese et al. :cite:`Neese2009`, but with more conservative grids and without P-junction screening. Overlap fitting is used to reduce aliasing errors :cite:`Izsak2011`. SGX scales as :math:`O(N^2)` with system size, as opposed to the :math:`O(N^4)` scaling of analytical exchange :cite:`Neese2009`.
+
+Usage and Example
+=================
+
+Any :attr:`scf.hf.SCF` object :attr:`mf` can be converted to an equivalent object that computes the Coulomb and Exchange matrices with SGX instead of analytical integration by calling :code:`sgx.sgx_fit(mf)`.
+
+* :source:`examples/sgx/00-simple_sgx.py`
+
+.. literalinclude:: ../../examples/sgx/00-simple_sgx.py
+
+.. code::
+
+  converged SCF energy = -76.0267374704185
+  converged SCF energy = -76.0267978618974
+
+In this case, the error of SGX compared to analytical exchange is about 0.06 mH. The line
+
+.. code::
+
+    mf.with_df.dfj = True
+
+specifies to compute the Coulomb matrix using Resolution of the Identity (RI) while using SGX for the Exchange matrix.
+
+Adjustable Parameters
+=====================
+
+Calling the :attr:`sgx_fit` function on an :attr:`scf.hf.SCF` object returns an equivalent :attr:`SGXHF` object with a :attr:`with_df` attribute that handles SGX integration. To use a non-default auxbasis (for :attr:`dfj=True`), :attr:`auxbasis` can be specified in the call to :attr:`sgx_fit`. In addition, there are five main adjustable parameters for this object:
+
+* :attr:`grids_level_i`: The level of a coarse grid to use for initial SCF iterations.
+* :attr:`grids_level_f`: The level of a denser grid to use for final SCF iterations.
+* :attr:`grids_thrd`: The grid points at which all atomic orbitals have a value below this threshold are removed from the integration grid.
```

Comment:
I would think this controls the threshold at which atomic orbitals are removed from consideration on the grid

Reply:
I think that would be a more efficient implementation, but that is not how it is currently implemented. See https://github.com/pyscf/pyscf/blob/7be5e015b2b40181755c71d888449db936604660/pyscf/sgx/sgx_jk.py#L96 and https://github.com/pyscf/pyscf/blob/7be5e015b2b40181755c71d888449db936604660/pyscf/sgx/sgx_jk.py#L329. I am going to do some optimization work on the SGX code itself in the future, but in the mean time I think the code only removes grid points if all AOs are below the threshold there. AOs are only screened by S-junction screening at the C level.

Reply:
Ohhh so that's what it means: grid points where no basis functions are significant are removed; this setting controls the significance cutoff.